src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesResource.kt

```diff
@@ -240,4 +244,240 @@ class IncentivesResource(private val incentivesService: IncentivesService) {
+    code: String,
+  ): IEPLevel =
+    incentivesService.getGlobalIncentiveLevel(code)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_INCENTIVES')")
+  @PostMapping("/incentives/reference-codes/reorder")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "reorders all global incentive levels",
+    description = "reorders all global incentive levels using provided list of Incentive codes, including inactive. 1-based index",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = ReorderRequest::class),
+        ),
+      ],
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Reorder successful",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden to access this endpoint when role NOMIS_INCENTIVES not present",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun reorderGlobalIncentiveLevels(
+    @RequestBody
+    request: ReorderRequest,
+  ) {
+    incentivesService.reorderGlobalIncentiveLevels(request.codeList)
+  }
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_INCENTIVES')")
+  @DeleteMapping("/incentives/reference-codes/{code}")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Deletes an existing global incentive level",
+    description = "Deletes an existing global incentive level, if level doesn't exist request is ignored",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Deletion successful (or ignored)",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden to access this endpoint when role NOMIS_INCENTIVES not present",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun deleteGlobalIncentiveLevel(
+    @Schema(description = "Incentive reference code", example = "STD", required = true)
+    @PathVariable
+    code: String,
+  ) {
+    incentivesService.deleteGlobalIncentiveLevel(code)
```

Comment:
I guess this is just for testing and would fail if a prisoner was on the level that needs to be deleted ?

Reply:
yes - just for testing